src/plugins/shared-variables/graph/plotted-variables-adornment/plotted-variables-adornment-model.test.ts

```diff
@@ -0,0 +1,167 @@
+    expect(model.plottedVariables.size).toBe(0);
+  });
+
+  it("should clear plotted variables", () => {
+    const model = PlottedVariablesAdornmentModel.create();
+    model.addPlottedVariables("a", "x", "y");
+    model.addPlottedVariables("b", "x", "y");
+    expect(model.plottedVariables.size).toBe(2);
+    model.clearPlottedVariables();
+    expect(model.plottedVariables.size).toBe(0);
+  });
+
+  it("should return variable values", () => {
+    const model = PlottedVariablesAdornmentModel.create();
+    model.addPlottedVariables("a", "x", "y");
+    const values = model.variableValues;
+    expect(values.x).toEqual([10]);
+    expect(values.y).toEqual([20]);
+  });
+
+  it("should return annotatable objects", () => {
+    const model = PlottedVariablesAdornmentModel.create();
+    model.addPlottedVariables("a", "x", "y");
+    const objects = model.annotatableObjects;
+    expect(objects.length).toBe(1);
+    expect(objects[0].objectId).toBe("var:{x}:{y}");
+    expect(objects[0].objectType).toBe("variable");
+  });
+
+  it("should get annotatable object position", () => {
+    const model = PlottedVariablesAdornmentModel.create();
+    model.addPlottedVariables("a", "x", "y");
+    const pos = model.getAnnotatableObjectPosition("variable", "var:{x}:{y}");
+    expect(pos).toEqual({ x: 10, y: 20 });
+    expect(model.getAnnotatableObjectPosition("other", "var:{x}:{y}")).toBeUndefined();
+  });
+
+  it("should return numeric values for attr role", () => {
+    const model = PlottedVariablesAdornmentModel.create();
+    model.addPlottedVariables("a", "x", "y");
+    const xVals = model.numericValuesForAttrRole("x");
+    expect(xVals).toContain(20); // 2 * 10
+    expect(xVals).toContain(0);
+    const yVals = model.numericValuesForAttrRole("y");
+    expect(yVals).toContain(40); // 2 * 20
+    expect(yVals).toContain(0);
+    expect(model.numericValuesForAttrRole("rightNumeric")).toEqual([]);
+  });
+
+  it("should return numeric values based on the graph axis if the variable doesn't have a value", () => {
+    sharedVariables = {
+      variables: [
+        { id: "x", name: "x" },
+        { id: "y", inputs: [ "x" ], expression: "x * 2" }
+      ]
+    };
+    const graph = GraphModel.create({});
+    const bottomAxis = NumericAxisModel.create({ place: "bottom", min: 0, max: 100 });
+    graph.setAxis("bottom", bottomAxis);
+    const model = PlottedVariablesAdornmentModel.create();
+    graph.addAdornment(model);
+    model.addPlottedVariables("a", "x", "y");
+    const xVals = model.numericValuesForAttrRole("x");
+    // with no value set for "x" the empty array is returned and the code using
+    // this function will just let th existing graph's x axis limits remain the same
+    expect(xVals).toEqual([]);
+    const yVals = model.numericValuesForAttrRole("y");
+    // The domain of the graph x axis is [100, 0], so the y values will be
+    // based on the limits of this domain
+    expect(yVals).toEqual([200, 0]);
```

Comment:
The domain is in the form `[max, min]`? I know that's not part of this PR, but I'm surprised it's not `[min, max]`.